crates/trigger-http/src/handler.rs

```diff
@@ -26,7 +26,7 @@ pub struct HttpHandlerExecutor;
 impl HttpExecutor for HttpHandlerExecutor {
     async fn execute(
         &self,
-        engine: &TriggerAppEngine<HttpTrigger>,
+        engine: &Arc<TriggerAppEngine<HttpTrigger>>,
```

Comment:
You don't often see `&Arc` params; typically you would use an owned `Arc` and let the caller decide whether it needs to clone or not.